### PR TITLE
chore(demo-theme): themed demo helper

### PR DESCRIPTION
**Pull request by alisonailea**
**Related Issue:** #9644

## Summary

themed demo helper

### CODE REVIEW

**Review by jcfranco — APPROVED** [review]
Awesome! 

Approving to get this in for today's testing, but I do have some notes:

* Can we add doc to the theme helper with setup examples?
* Are the updates to the button page temporary or something that we'd like to promote across all component pages? We could also consider adding a specific theme demo page. 
* Can we add a note on current limitations of the helper? IIRC, stateful CSS props are not supported.

File: packages/calcite-components/src/demos/_assets/demo-theme.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
+const defaultTheme = {
+  background: "blue",
+  cornerRadius: "42px",
+  shadowColor: "black",
+  shadowValues: "0 1px 2px 0 1",
+  space: "24px",
+  text: "green",
```
Comment (driskull) [inline packages/calcite-components/src/demos/_assets/demo-theme.ts:7]: should this maybe be `textColor`?

**Review by driskull — APPROVED** [review]
👍

**Comment by alisonailea** [comment]
> * Can we add doc to the theme helper with setup examples?
Added

> * Are the updates to the button page temporary or something that we'd like to promote across all component pages? We could also consider adding a specific theme demo page.

Not intended to be temporary. The idea was to allow these demos to be added to any page although long-term it might be good to have a specific theme demo page with all/many components.

> * Can we add a note on current limitations of the helper? IIRC, stateful CSS props are not supported.
There aren't any limitations as far as I know. Stateful CSS props are supported.

**Pull request merged**